x-pack/metricbeat/module/containerd/cpu/cpu.go

```diff
@@ -120,82 +162,97 @@ func (m *metricset) Fetch(reporter mb.ReporterV2) error {
 		}
 	}
 
-	for _, event := range events {
-		// setting ECS container.id and module field containerd.namespace
+	// Iterate through each group and consolidate them into a single event per group
+	for _, group := range grouped {
+		cID := containerd.GetAndDeleteCid(group[0])
+		for _, event := range events {
```

Comment:
Just to check, should these events be the ones that are part of the group we are iterating over, and not all events? @gpop63

Reply:
It should range over `group` instead of `events`. Great catch!